core/kotlinx-coroutines-core/service-loader-test/test/ServiceLoaderTest.kt

```diff
@@ -0,0 +1,32 @@
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.internal.FastServiceLoader
+import kotlinx.coroutines.internal.MainDispatcherFactory
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ServiceLoaderTest {
+
+    @InternalCoroutinesApi
+    @Test
+    fun testLoadingCrossModuleService() {
+        val providers = MainDispatcherFactory::class.java.let { FastServiceLoader.loadProviders(it, it.classLoader) }
+        assertEquals(providers.size, 2)
```

Comment:
It is better to verify actual classes as well

Reply:
How can I verify that?

Reply:
E.g. by class name